File: insertion_sort.py
def insertion_sort(ar, start, end):
    if start >= end:
        return

    else:

        curPos = temp = 0

        for x in range(start + 1, end):
            curPos = x

            while curPos > 0 and (ar[curPos] < ar[curPos - 1]):

                temp = ar[curPos]
                ar[curPos] = ar[curPos - 1]
                ar[curPos - 1] = temp

                curPos -= 1


n = int(raw_input("Enter the no. of elements "))

ar = []

for x in range(n):
    ar.append(int(raw_input("Enter element " + str(x + 1) + " ")))

insertion_sort(ar, 0, len(ar))

print ar

File: CPP/Algorithms/InsertionSort.cpp
#include <iostream>
using namespace std;
void insertionSort(int ar[], int start, int end)
{
    if (start >= end)
        return;
    else
    {
        int curPos, temp;
        curPos = temp = 0;
        for (int i = start + 1; i < end; i++)
        {
            curPos = i;
            while (curPos > 0 && (ar[curPos] < ar[curPos - 1]))
            {
                temp = ar[curPos];
                ar[curPos] = ar[curPos - 1];
                ar[curPos - 1] = temp;

                curPos -= 1;
            }
        }
    }
}

// Test Code

int main()
{
    int n;
    cin >> n;
    int ar[n];
    for (int i = 0; i < n; i++)
        cin >> ar[i];
    insertionSort(ar, 0, n);
    for (int i = 0; i < n; i++)
        cout << ar[i] << ' ';
    cout << endl;
    return 0;
}